certbot-ci/certbot_integration_tests/certbot_tests/context.py

```diff
@@ -1,16 +1,134 @@
+"""Module do handle the context of integration tests."""
+import os
+import tempfile
+import subprocess
+import shutil
+import sys
+from distutils.version import LooseVersion
+
+from certbot_integration_tests.utils import misc
+
+
 class IntegrationTestsContext(object):
     """General fixture describing a certbot integration tests context"""
     def __init__(self, request):
         self.request = request
+
+        self.worker_id = request.config.slaveinput['slaveid'] if hasattr(request.config, 'slaveinput') else 'master'
         if hasattr(request.config, 'slaveinput'):  # Worker node
             self.worker_id = request.config.slaveinput['slaveid']
             self.acme_xdist = request.config.slaveinput['acme_xdist']
         else:  # Primary node
             self.worker_id = 'primary'
             self.acme_xdist = request.config.acme_xdist
+
+        self.acme_server =self.acme_xdist['acme_server']
         self.directory_url = self.acme_xdist['directory_url']
         self.tls_alpn_01_port = self.acme_xdist['https_port'][self.worker_id]
         self.http_01_port = self.acme_xdist['http_port'][self.worker_id]
+        self.challtestsrv_mgt_port = self.acme_xdist['challtestsrv_port']
+
+        self.certbot_version = misc.get_certbot_version()
```

Comment:
This variable is used only in `_common_test_no_force_renew`, but `_common_test_no_force_renew` can be called a lot of time in the same test. Here I cache the certbot version value to avoid to invoke certbot in a subprocess each time `_common_test_no_force_renew` is called.

Reply:
Great explanation, that would make a great comment on why this line is here in the code.

Reply:
Done.